src/main/java/org/asamk/signal/commands/SendCommand.java

```diff
@@ -46,7 +46,7 @@ public int handleCommand(final Namespace ns, final Signal signal) {
             return 1;
         }
 
-        if (ns.getList("recipient") == null || ns.getList("recipient").size() == 0) {
+        if ((ns.getList("recipient") == null || ns.getList("recipient").size() == 0) && ns.getString("group") == null) {
```

Comment:
Thanks for fixing this!
Can you please move the check for empty recipient into the ` if (ns.getBoolean("endsession")) {`, because that case doesn't support groups.
And check for recipient or group below that `if` block.

Reply:
How's this? I don't like doing redundant checks, but I couldn't think of a better way to do it.

Reply:
Hm, this doesn't show an error if groupid and endsession are set and no recipients, but instead tries to send a group message in that case.

How about this check:
```

		        if ((ns.getList("recipient") == null || ns.getList("recipient").size() == 0) && (ns.getBoolean("endsession") || ns.getString("group") == null)) {
		            System.err.println("No recipients given");
		            System.err.println("Aborting sending.");
		            return 1;
		        }```